src/Development/IDE/Core/Compile.hs

```diff
@@ -247,10 +250,55 @@ upgradeWarningToError (nfp, sh, fd) =
   warn2err = T.intercalate ": error:" . T.splitOn ": warning:"
 
 hideDiag :: DynFlags -> (WarnReason, FileDiagnostic) -> (WarnReason, FileDiagnostic)
-hideDiag originalFlags (Reason warning, (nfp, _sh, fd))
-  | not (wopt warning originalFlags) = (Reason warning, (nfp, HideDiag, fd))
+hideDiag originalFlags (Reason warning, (nfp, sh, fd))
+  | not (wopt warning originalFlags)
+  = if warning `elem` unnecessaryDeprecationWarningFlags
+       then (Reason warning, (nfp, sh, fd{_severity = Just DsInfo}))
+       else (Reason warning, (nfp, HideDiag, fd))
 hideDiag _originalFlags t = t
 
+enableUnnecessaryAndDeprecationWarnings :: ParsedModule -> ParsedModule
+enableUnnecessaryAndDeprecationWarnings =
+  (update_pm_mod_summary . update_hspp_opts)
+  (foldr (.) id [(`wopt_set` flag) | flag <- unnecessaryDeprecationWarningFlags])
+
+-- warnings which lead to a diagnostic tag
+unnecessaryDeprecationWarningFlags :: [WarningFlag]
+unnecessaryDeprecationWarningFlags
+  = [ Opt_WarnUnusedTopBinds
+    , Opt_WarnUnusedLocalBinds
+    , Opt_WarnUnusedPatternBinds
+    , Opt_WarnUnusedImports
+    , Opt_WarnUnusedMatches
+    , Opt_WarnUnusedTypePatterns
+    , Opt_WarnUnusedForalls
+#if MIN_GHC_API_VERSION(8,10,0)
+    , Opt_WarnUnusedRecordWildcards
+#endif
+#if MIN_GHC_API_VERSION(8,6,0)
+    , Opt_WarnInaccessibleCode
+#endif
+    , Opt_WarnWarningsDeprecations
+    ]
+
+tagDiag :: (WarnReason, FileDiagnostic) -> (WarnReason, FileDiagnostic)
```

Comment:
Could we get a comment on what this does and why? (From the code, its clear, but nice to be able to eyeball it without reading the code)

Reply:
Added comment to the file:

> -- | Add a unnecessary/deprecated tag to the required diagnostics.